src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -732,7 +764,7 @@ fn resolve_associated_trait_item(
     item_name: Symbol,
     ns: Namespace,
     cx: &mut DocContext<'_>,
-) -> Option<(ty::AssocKind, DefId)> {
+) -> Option<ty::AssocItem> {
```

Comment:
This change is not strictly necessary, but it seems like a good refactoring. The reason I made the change is that I was going to make `UrlFragment::from_assoc_item` take a `ty::AssocItem` and use that to get the defaultness, but that caused some tests to fail.